main/src/main/java/cgeo/geocaching/models/Geocache.java

```diff
@@ -1859,10 +1868,26 @@ public void changeVariables(final Map<String, String> newVars) {
         for (Map.Entry<String, String> var : newVars.entrySet()) {
             getVariables().addVariable(var.getKey(), var.getValue());
             getVariables().saveState();
-            getVariables().recalculateWaypoints(this); //need to do this because new wps are not saved to DB yet
+            recalculateWaypoints(); //need to do this because new wps are not saved to DB yet
         }
     }
 
+    public boolean recalculateWaypoints() {
+        return recalculateWaypoints(getVariables());
+    }
+
+    public boolean recalculateWaypoints(final CacheVariableList variableList) {
+        boolean hasCalculatedWp = false;
+        for (Waypoint wp : getWaypoints()) {
+            hasCalculatedWp |= wp.recalculateVariableDependentValues(variableList);
+        }
+        if (hasCalculatedWp) {
+            resetFinalDefined();
+            DataStore.saveFinalDefinedStatus(this);
+            DataStore.saveWaypoints(this);
```

Comment:
Maybe `saveFinalDefined` logic should be included in `saveWaypoints` method?

Reply:
Merged the new function into saveWaypoint